quic/transport/ngtcp2/connection/closedstate.nim

```diff
@@ -29,7 +29,7 @@ method openStream(
 ): Future[Stream] {.async: (raises: [CancelledError, ConnectionError, QuicError]).} =
   raise newException(ClosedConnectionError, "connection is closed")
 
-method close(state: ClosedConnection) {.async: (raises: []).} =
+method close(state: ClosedConnection) {.async: (raises: [CancelledError, QuicError]).} =
```

Comment:
- would it be possible to have `{.async: (raises: []).}` for `close` method?
- should it really propagate `CancelledError`? closing resources shouldn't take much time?
- what should user do if any of these error is raised?

Reply:
- propagating `CancelledError` might be okay, but i am not sure if `QuicError` should be propagated 
  - `noCancel` could be added to prevent cancellation of `close` by users. so they can have flexibility to choose what they want to do
  - if `QuicError` happens, what should user do? just log exception?

Reply:
after thinking this through, it's fine to raise exceptions :)